pkg/kubectl/cmd/annotate.go

```diff
@@ -193,7 +193,11 @@ func (o AnnotateOptions) RunAnnotate() error {
 			return err
 		}
 
-		name, namespace, obj := info.Name, info.Namespace, info.Object
+		obj, err := resource.AsVersionedObject([]*resource.Info{info}, false, info.Mapping.GroupVersionKind.Version)
```

Comment:
hmm, maybe not... looks like VersionedObject is nil when loading from the server